### PR TITLE
Ensure Oxo serve & RE_Oxo schemas are similar.

### PR DESCRIPTION
**Pull request by deadly-panda**

### CODE REVIEW

File: tests/serve_app/oxo_test.py [hunk 1]
```diff
@@ -1667,3 +1674,360 @@ def testDeleteAgentGroupMutation_whenAgentGroupDoesNotExist_returnErrorMessage(
...
+
+
+def _get_re_oxo_schema(query: str) -> Dict[str, Any]:
+    """Introspect the oxo endpoint on RE, and fetch desired definitions."""
+    with httpx.Client(verify=True) as client:
+        response = client.post(
+            RE_OXO_ENDPOINT,
+            data={"query": query},
+            timeout=10,
+            headers={"X-Api-Key": f"{os.environ.get('RE_OXO_API_KEY', "")}"},
```
Comment (deadly-panda) [inline tests/serve_app/oxo_test.py:1686]:
```suggestion
            headers={"X-Api-Key": f"{os.environ.get('RE_OXO_API_KEY', '')}"},
```

File: tests/serve_app/oxo_test.py [hunk 2]
```diff
@@ -2346,3 +2353,507 @@ def testRunScanMutation_whenAssetDoesNotExist_returnErrorMessage(
...
+        }    
+    """
+
+    oxo_schema_dict = oxo_schema.introspect()["__schema"]
+
+    re_oxo_schema_dict = _get_re_oxo_schema(introspect_enums_query)
+
+    re_oxo_types = re_oxo_schema_dict["__schema"]["types"]
+
+    re_oxo_enums = {}
```
Comment (3asm) [inline tests/serve_app/oxo_test.py:2396]: Please group lines to have:

setup

run

assert.

File: tests/serve_app/oxo_test.py [hunk 3]
```diff
@@ -2346,3 +2353,507 @@ def testRunScanMutation_whenAssetDoesNotExist_returnErrorMessage(
...
+        )
+
+        return response.json()["data"]
+
+
+@pytest.mark.skip(reason="Schema not complete on RE_OXO.")
+def testOxoSchemaReOxoSchemas_whenEnums_schemasShouldBeSimilar() -> None:
+    """Ensure the `ENUMs` in the OxO Schema & RE_OxO schema are similar."""
+
+    introspect_enums_query = """
```
Comment (3asm) [inline tests/serve_app/oxo_test.py:2376]: CONST?

File: tests/serve_app/oxo_test.py [hunk 4]
```diff
@@ -2346,3 +2353,507 @@ def testRunScanMutation_whenAssetDoesNotExist_returnErrorMessage(
...
+
+    oxo_types = oxo_schema_dict["types"]
+
+    oxo_query_fields = [
+        t for t in oxo_types if t["kind"] == "OBJECT" and t["name"] == "Query"
+    ][0]["fields"]
+
+    oxo_queries = {query["name"]: query for query in oxo_query_fields}
+
+    for query_name, queriy_fields in re_oxo_queries.items():
```
Comment (3asm) [inline tests/serve_app/oxo_test.py:2666]: Can you please add assert that the list is not empty.

**Review by 3asm — CHANGES_REQUESTED** [review]

**Review by benyissa — COMMENTED** [review]
Do we need to always run this check? Why not put it in a separate workflow and trigger it manually when needed?

**Comment by 3asm** [comment]
Ping @deadly-panda on PR fixed.

**Review by 3asm — APPROVED** [review]

**Pull request merged**